modules/platforms/cpp/ignite/client/detail/node_connection.cpp

```diff
@@ -203,7 +208,46 @@ std::shared_ptr<response_handler> node_connection::find_handler_unsafe(std::int6
     if (it == m_request_handlers.end())
         return {};
 
-    return it->second;
+    return it->second.handler;
+}
+
+void node_connection::handle_timeouts() {
+    auto now = std::chrono::steady_clock::now();
+
+    {
+        std::lock_guard lock(m_request_handlers_mutex);
+
+        std::vector<int64_t> keys_for_erasure;
```

Comment:
Let's try avoiding additional memory allocation here. Let's erase values as we move through the map. We hold the lock anyway. It can be done with iterator:
```cpp
auto it = map.begin();
while (it != map.end()) {
  if (something) {
    it = map.erase(it);
    continue;
  }
  ++it;
}
```